tss-esapi/src/tcti_ldr.rs

```diff
@@ -356,35 +383,58 @@ fn validate_from_str_device_config() {
 ///
 /// The default configuration will point to `localhost:2321`
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct NetworkTPMConfig {
-    /// Address of the server to connect to
-    ///
-    /// Defaults to `localhost`
-    host: ServerAddress,
-    /// Port used by the server at the address given in `host`
-    ///
-    /// Defaults to `2321`
-    port: u16,
+pub enum TpmSimulatorConfig {
+    /// Connects to tpm over TCP
+    Tcp {
+        /// Address of the server to connect to
+        ///
+        /// Defaults to `localhost`
+        host: ServerAddress,
+        /// Port used by the server at the address given in `host`
+        ///
+        /// Defaults to `2321`
+        port: u16,
+    },
+    #[cfg(has_simulator_unix_socket)]
+    /// Connects to tpm over Unix socket
+    Unix { path: String },
 }
 
-const DEFAULT_SERVER_PORT: u16 = 2321;
+impl TpmSimulatorConfig {
+    const DEFAULT_PORT_CONFIG: u16 = 2321;
+}
 
-impl Default for NetworkTPMConfig {
+impl Default for TpmSimulatorConfig {
     fn default() -> Self {
-        NetworkTPMConfig {
+        TpmSimulatorConfig::Tcp {
             host: Default::default(),
-            port: DEFAULT_SERVER_PORT,
+            port: TpmSimulatorConfig::DEFAULT_PORT_CONFIG,
         }
     }
 }
 
-impl FromStr for NetworkTPMConfig {
+impl FromStr for TpmSimulatorConfig {
     type Err = Error;
 
     fn from_str(config_str: &str) -> Result<Self> {
         if config_str.is_empty() {
             return Ok(Default::default());
         }
+
+        #[cfg(has_simulator_unix_socket)]
+        {
+            let path_pattern = Regex::new(r"(,|^)path=(.*?)(,|$)").unwrap(); // should not fail
+            if let Some(path) = path_pattern
+                .captures(config_str)
+                .and_then(|c| c.get(2))
+                .map(|m| m.as_str())
+            {
+                return Ok(TpmSimulatorConfig::Unix {
+                    path: path.to_string(),
+                });
+            }
+        }
```

Comment:
Hmm, this means that the Unix socket has "priority" (i.e., if you have `,port=1234,host=127.0.0.1,path=/tmp/sock`, it'll use the Unix socket configuration), but the TCP socket is treated as default, if I understand correctly. This should probably be documented somewhere, and also tested.

Is this also an explicit choice upstream, or is it at most coincidental?

Reply:
(sorry I missed the notification)

It was just easier for me to implement that way, the upstream just parse it so that the last of `host` or `path` has priority and nullify the other.

Reply:
Sorry, I drifted away as well...

That sounds like a reasonable argument for doing it this way, I think my only request would be to add a couple of tests for this ordering, and maybe a line in the doc of `TpmSimulatorConfig`.

Reply:
@ionut-arm just to be clear, you expect me to replicate the behavior of upstream, correct?

Reply:
Ah, I misread that, apologies, thought your approach was identical to upstream. Hmm, well this is frustrating, would've been nice if the upstream project exposed functionality to parse this so we could all align.

But, since this is about a malign configuration, I wouldn't put too much effort in trying to implement the same treatment everywhere. Users really shouldn't expect an arguably wrong input to be parsed in some particular way.

Nah, just go ahead and document it, no need to change anything. Sorry again for the confusion.